src/crimson/osd/main.cc

```diff
@@ -187,14 +187,20 @@ int main(int argc, const char* argv[])
           crimson::net::MessengerRef cluster_msgr, client_msgr;
           crimson::net::MessengerRef hb_front_msgr, hb_back_msgr;
           for (auto [msgr, name] : {make_pair(std::ref(cluster_msgr), "cluster"s),
-                                    make_pair(std::ref(client_msgr), "client"s),
-                                    make_pair(std::ref(hb_front_msgr), "hb_front"s),
+                                    make_pair(std::ref(client_msgr), "client"s)}) {
+            msgr = crimson::net::Messenger::create(entity_name_t::OSD(whoami),
+                                                   name,
+                                                   nonce,
+                                                   false);
```

Comment:
When dispatch_only_on_primary_sid is set to true, the messenger was single-sharded -- it only dispatches connections in the same shard where the messenger is created.

When dispatch_only_on_primary_sid is set to false, the messenger becomes multiple-sharded -- it will dispatch connections (to the OSD) from all the available shards.